### PR TITLE
New Dashboard: Fixes for pause popup

### PR DESCRIPTION
**Pull request by yankurk**
Issues fixed in this PR related to pause popup in the new dashboard:
- Auto focus on the text field when the popup opens
- Trigger pause and close popup when enter is pressed in the text field
- Close popup upon pressing escape key
- Do not retain pause message entered once the popup is closed

### CODE REVIEW

**Comment by yankurk** [comment]
#4298

**Comment by ketan** [comment]
This does not work as expected when the modal is closed and re-opened the second time.

untitled

File: server/webapp/WEB-INF/rails.new/webpack/views/shared/new_modal.js.msx [hunk 1]
```diff
@@ -55,11 +57,27 @@ const Modal = function (options) {
           destroy();
         }
       };
+
+      this.handleEnterKey = (e) => {
+        if (e.keyCode === ENTER_KEY_CODE) {
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/webpack/views/shared/new_modal.js.msx]: this has been deprecated https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

**Comment by yankurk** [comment]
Fixes #4309

**Comment by ketan** [comment]
@yankurk — can you rebase and confirm if this is ready to review and merge?

**Comment by yankurk** [comment]
@ketan I have rebased the PR. It needs to be reviewed.

**Pull request merged**